### ISSUE TITLE
Accelerated reports rate calculations bug.

### ISSUE THREAD

**Issue by pblo** [afarran/swm2m_vms_tests#33]
Hi, 

Quoting documentation:

Example:
- If VMS service start-up time or configuration change time was 0,
- And if StandardReport1Interval is 60 minutes,
- and AcceleratedReport1Rate is 6.
- Then StandardReport1 is sent at minute 60,
- and AcceleratedReport1 is sent at minute 10, 20, 30, 40, 50.

but when I configure TC as follows:

StandardReport1Interval = 1min (60secs)
AcceleratedReport1Rate = 6 

then only StandardReport1 is sent (every 60secs) several times, with only occasional Accelerated Reports (with wrong timeouts)

These are exact details about timing:

ConfigChangeReport 15:18:32
StandardReport1 15:19:31
StandardReport1 15:20:31
StandardReport1 15:21:31
StandardReport1 15:22:31
StandardReport1 15:23:31
StandardReport1 15:24:30
AcceleratedReport1 15:24:31 (wrong!)
StandardReport1 15:25:30
AcceleratedReport1 15:25:31 (wrong!)
StandardReport1 15:26:31
StandardReport1 15:27:31
StandardReport1 15:28:31
StandardReport1 15:29:31
StandardReport1 15:30:31

Which is not the desired behaviour described in documentation...

Debug information:

shell> prop get VMS
SIN=115(VMS) PIN=1(LogReportInterval) val=2
SIN=115(VMS) PIN=2(StandardReport1Interval) val=1
SIN=115(VMS) PIN=3(AcceleratedReport1Rate) val=6
SIN=115(VMS) PIN=4(StandardReport2Interval) val=0
SIN=115(VMS) PIN=5(AcceleratedReport2Rate) val=1
SIN=115(VMS) PIN=6(StandardReport3Interval) val=0
SIN=115(VMS) PIN=7(AcceleratedReport3Rate) val=1
SIN=115(VMS) PIN=30(GpsJammedSendReport) val=true
SIN=115(VMS) PIN=31(GpsJammedStartDebounceTime) val=1
SIN=115(VMS) PIN=32(GpsJammedEndDebounceTime) val=1
SIN=115(VMS) PIN=33(GpsJammedState) val=false
SIN=115(VMS) PIN=34(GpsBlockedSendReport) val=true
SIN=115(VMS) PIN=35(GpsBlockedStartDebounceTime) val=1
SIN=115(VMS) PIN=36(GpsBlockedEndDebounceTime) val=1
SIN=115(VMS) PIN=37(GpsBlockedState) val=false
SIN=115(VMS) PIN=38(IdpBlockedSendReport) val=true
SIN=115(VMS) PIN=39(IdpBlockedStartDebounceTime) val=1
SIN=115(VMS) PIN=40(IdpBlockedEndDebounceTime) val=1
SIN=115(VMS) PIN=41(IdpBlockedState) val=false
SIN=115(VMS) PIN=42(HwClientDisconnectedSendReport) val=true
SIN=115(VMS) PIN=43(HwClientDisconnectedStartDebounceTime) val=1
SIN=115(VMS) PIN=44(HwClientDisconnectedEndDebounceTime) val=1
SIN=115(VMS) PIN=45(HwClientDisconnectedState) val=false
SIN=115(VMS) PIN=46(InterfaceUnitDisconnectedSendReport) val=true
SIN=115(VMS) PIN=47(InterfaceUnitDisconnectedStartDebounceTime) val=120
SIN=115(VMS) PIN=48(InterfaceUnitDisconnectedEndDebounceTime) val=120
SIN=115(VMS) PIN=49(InterfaceUnitDisconnectedState) val=false
SIN=115(VMS) PIN=50(InsideGeofenceSendReport) val=false
SIN=115(VMS) PIN=51(InsideGeofenceStartDebounceTime) val=0
SIN=115(VMS) PIN=52(InsideGeofenceEndDebounceTime) val=0
SIN=115(VMS) PIN=53(InsideGeofenceState) val=false
SIN=115(VMS) PIN=54(PowerDisconnectedSendReport) val=true
SIN=115(VMS) PIN=55(PowerDisconnectedStartDebounceTime) val=1
SIN=115(VMS) PIN=56(PowerDisconnectedEndDebounceTime) val=1
SIN=115(VMS) PIN=57(PowerDisconnectedState) val=false
SIN=115(VMS) PIN=100(PropertyChangeDebounceTime) val=1
SIN=115(VMS) PIN=101(MinStandardReportLedFlashTime) val=1
SIN=115(VMS) PIN=102(ShellTimeout) val=5
SIN=115(VMS) PIN=103(MailSessionIdleTimeout) val=5
SIN=115(VMS) PIN=104(GpsInEmails) val=true
SIN=115(VMS) PIN=105(AllowedEmailDomains) val=

Version:

IdpPackage: = "6.0.10.9363"
VmsAgent = "1.4.1"
InterfaceUnit = "2.2.5"

BR, Paweł

**Comment by afarran** [comment]
Please post the trace output log that shows when this happens.

**Comment by pblo** [comment]
Yes. I will replicate it one more time and paste trace output soon.

**Comment by pblo** [comment]
Here you have the trace output and screenshot of OTA messages timing:

https://github.com/afarran/swm2m_vms_tests/tree/dev/TraceOutputs/Issue%2333

pblo

**Comment by pblo** [comment]
We also checked this issue on idp-680 and with the same setup only standard reports are comming.

The issue only exists when accelerated reports timeout is calculated to value less than 60 secs.

That is interesting part of the trace log:

```
[   221.621:22:ltrc] [115:INFO] Report 1. Next Std 60s Acc 60s

```

With this setup Acc shouldn't be 60s , should be 10s.

More trace of the idp-680 is here: https://github.com/afarran/swm2m_vms_tests/blob/dev/TraceOutputs/Issue%2333/trace680.log

When we set other values:

StandardReport1Interval = 2min (120secs)
AcceleratedReport1Rate = 2

then VMS service is logging:

[   221.621:22:ltrc] [115:INFO] Report 1. Next Std 120s Acc 60s

and indeed is sending messages like that.

So the issue only concerns Acc timeouts less than 60s , like this one from example in documentation.

**Comment by afarran** [comment]
No Standard/Accelerated report should have an interval less than 1 minute. Anything calculated to less than a minute will be rounded up to 1 minute.
There's an explicit note on this in the "Report" module of the documentation:

> The minimum interval between 2 position reports from same reporting service is 1 minute.

**Comment by pblo** [comment]
Ok. I thought in seconds instead minutes in the example, however there is still something wrong.

The timeout is rounded to a minute, but in this example standard report has 1 minute as well, so accelerated report should not come at all and it is comming:

ConfigChangeReport 15:18:32
StandardReport1 15:19:31
StandardReport1 15:20:31
StandardReport1 15:21:31
StandardReport1 15:22:31
StandardReport1 15:23:31
StandardReport1 15:24:30
AcceleratedReport1 15:24:31 (wrong!)
StandardReport1 15:25:30
AcceleratedReport1 15:25:31 (wrong!)
StandardReport1 15:26:31
StandardReport1 15:27:31
StandardReport1 15:28:31
StandardReport1 15:29:31
StandardReport1 15:30:31

Sometimes accelerated report comes one second after standard report, so it is kind of bug anyway..

**Comment by afarran** [comment]
Yes, it seems like a bug. Our Verification team who test with real terminals never saw that, so I wonder if it is due to how fast Terminal Simulator runs. I'll consider this bug minor if we can't replicate it on a Desktop/real terminal.

**Comment by pblo** [comment]
We didn't replicate that on real device so far ..
